File: program51.cpp
#include<iostream>
using namespace std;

class Digit
{
    public:
        int iNo = 0;

    Digit(int A)
    {
        iNo = A;
    }
    void DisplayDigit()
    {
        int iDigit = 0;

        while(iNo > 0)
        {
            iDigit = iNo % 10;
            cout<<iDigit<<endl;
            iNo = iNo / 10;
        }
    }
};



int main()
{
    int iValue = 0;

    cout<<"Enter the number :"<<endl;
    cin>>iValue;

    Digit dobj(iValue);
    dobj.DisplayDigit();

    return 0;
}

File: program51.java

import java.util.*;

class Digit
{
    public int iNo;

    public Digit(int A)
    {
        iNo = A;
    }
    public void DisplayDigit()
    {
        int iDigit = 0;

        while(iNo > 0)
        {
            iDigit = iNo % 10;
            System.out.println(+iDigit);
            iNo = iNo / 10;
        }
    }
}

class program51
{
    public static void main(String Arg[])
    {
        Scanner obj = new Scanner(System.in);
        int iValue = 0;

        System.out.println("Enter the number :");
        iValue = obj.nextInt();

        Digit dobj = new Digit(iValue);
        dobj.DisplayDigit();
    }
}